components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java

```diff
@@ -403,48 +444,105 @@ private ResultSet getIdpQueryResultSet(Connection dbConnection, String sortedOrd
-     * @param dbConnection database Connection.
+     * @param resultSet          ResultSet.
+     * @param dbConnection       Database Connection.
+     * @param requiredAttributes Required attributes which needs to be return.
+     * @param tenantId           Tenant Id of the identity provider.
      * @return List of Identity Provider.
      * @throws SQLException Database Exception.
      */
-    private List<IdentityProvider> populateIdentityProviderList(ResultSet resultSet, Connection dbConnection)
-            throws SQLException {
+    private List<IdentityProvider> populateIdentityProviderList(ResultSet resultSet, Connection dbConnection,
+                                                                List<String> requiredAttributes, int tenantId)
+            throws SQLException, IdentityProviderManagementServerException {
 
         List<IdentityProvider> identityProviderList = new ArrayList<>();
         while (resultSet.next()) {
             IdentityProvider identityProvider = new IdentityProvider();
-            identityProvider.setIdentityProviderName(resultSet.getString(1));
-            if ((IdPManagementConstants.IS_TRUE_VALUE).equals(resultSet.getString("IS_PRIMARY"))) {
-                identityProvider.setPrimary(true);
-            } else {
-                identityProvider.setPrimary(false);
-            }
-            identityProvider.setHomeRealmId(resultSet.getString("HOME_REALM_ID"));
+            // First set the basic attributes such as id, name, description, isEnabled, image url, uuid.
+            identityProvider.setId(resultSet.getString("ID"));
+            identityProvider.setIdentityProviderName(resultSet.getString("NAME"));
             identityProvider.setIdentityProviderDescription(resultSet.getString("DESCRIPTION"));
-            // IS_FEDERATION_HUB_IDP
-            if ((IdPManagementConstants.IS_TRUE_VALUE).equals(resultSet.getString("IS_FEDERATION_HUB"))) {
-                identityProvider.setFederationHub(false);
-            }
-            // IS_LOCAL_CLAIM_DIALECT
-            if ((IdPManagementConstants.IS_TRUE_VALUE).equals(resultSet.getString("IS_LOCAL_CLAIM_DIALECT"))) {
-                if (identityProvider.getClaimConfig() == null) {
-                    identityProvider.setClaimConfig(new ClaimConfig());
-                }
-                identityProvider.getClaimConfig().setLocalClaimDialect(true);
-            }
             // IS_ENABLE
             if ((IdPManagementConstants.IS_TRUE_VALUE).equals(resultSet.getString("IS_ENABLED"))) {
                 identityProvider.setEnable(true);
             } else {
                 identityProvider.setEnable(false);
             }
-            identityProvider.setDisplayName(resultSet.getString("DISPLAY_NAME"));
             identityProvider.setImageUrl(resultSet.getString("IMAGE_URL"));
             identityProvider.setResourceId(resultSet.getString("UUID"));
+
+            if (requiredAttributes != null) {
```

Comment:
do we need this null check since we are already checking it at line 575

Reply:
fixed